File: frontend/src/api/dummy.ts
import type { components } from './types.generated'

type BundleInfo = components['schemas']['BundleInfo']
type Document = components['schemas']['Document']

export async function listBundles(): Promise<BundleInfo[]> {
  const resp = await fetch('/api/dummy/bundles')
  if (!resp.ok) throw new Error('번들 목록을 불러오지 못했습니다.')
  return resp.json() as Promise<BundleInfo[]>
}

export async function loadBundle(name: string): Promise<Document[]> {
  const resp = await fetch(`/api/dummy/load/${encodeURIComponent(name)}`, { method: 'POST' })
  if (!resp.ok) {
    const err = await resp.json().catch(() => ({}))
    throw new Error((err as { detail?: string }).detail ?? '번들 적재에 실패했습니다.')
  }
  return resp.json() as Promise<Document[]>
}

File: backend/app/api/dummy.py
"""Sample bundle loader endpoints for demo/QA purposes."""
from __future__ import annotations

from fastapi import APIRouter, HTTPException, status

from app.models.api import BundleInfo
from app.models.document import Document
from app.services.dummy_service import list_bundles, load_bundle

router = APIRouter(prefix="/api/dummy", tags=["Dummy"])


@router.get(
    "/bundles",
    response_model=list[BundleInfo],
    summary="List available sample bundles",
)
async def get_bundles() -> list[BundleInfo]:
    return [BundleInfo(**b) for b in list_bundles()]


@router.post(
    "/load/{bundle_name}",
    response_model=list[Document],
    status_code=status.HTTP_200_OK,
    summary="Reset state and load a sample bundle",
)
async def load(bundle_name: str) -> list[Document]:
    try:
        return load_bundle(bundle_name)
    except KeyError:
        raise HTTPException(
            status_code=status.HTTP_404_NOT_FOUND,
            detail=f"번들을 찾을 수 없습니다: {bundle_name}",
        )
    except (FileNotFoundError, ValueError) as exc:
        raise HTTPException(
            status_code=status.HTTP_422_UNPROCESSABLE_CONTENT,
            detail=str(exc),
        ) from exc
